Algorithms/Numeric/Factorial.cs

```diff
@@ -22,5 +23,37 @@ public static long Calculate(int num)
 
             return num == 0 ? 1 : num * Calculate(num - 1);
         }
+
+        /// <summary>
+        ///     Calculates factorial of a any integer number that implements IBinaryInteger.
+        /// </summary>
+        /// <param name="num">Any integer that implements IBinaryInteger.</param>
+        /// <typeparam name="T">The type of input integer.</typeparam>
+        /// <returns>Factorial of integer number.</returns>
+        public static BigInteger Calculate<T>(T num) where T : notnull, IBinaryInteger<T>
```

Comment:
Let's make it `T Calculate<T>(int n)`. The result should be generic, and the parameter can be int IMO because it's unreasonable to calculate factorials of bigger numbers.

Reply:
1. I will have to perform casting based on generic type T as follows for suggested method signature in order to make multiplication compatible in factorial
` 
T inputNum = (T)Convert.ChangeType(num, typeof(T));
`
Problem is 'BigInteger' doesn't implement 'IConvertible' Inteface and will throw exception for for above statement.
Thus my recommendation is to keep return type as 'BigInteger' and user will cast it as per their needs.
Please let me know your opinion on this.